include/lsst/pex/exceptions/python/Exception.h

```diff
@@ -0,0 +1,84 @@
+// -*- LSST-C++ -*-
+/* 
+ * LSST Data Management System
+ * Copyright 2008-2016  AURA/LSST.
+ * 
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the LSST License Statement and 
+ * the GNU General Public License along with this program.  If not, 
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_PEX_EXCEPTIONS_PYTHON_EXCEPTION_H
+#define LSST_PEX_EXCEPTIONS_PYTHON_EXCEPTION_H
+
+#include "lsst/pex/exceptions/Exception.h"
+
+#include <pybind11/pybind11.h>
+
+#include <ostream>
+#include <string>
+
+namespace lsst {
+namespace pex {
+namespace exceptions {
+namespace python {
+
+/** Helper function for pybind11, used to define new types of exceptions.
+  * \param[in] mod Module to insert the exception into.
+  * \param[in] name Name of the exception in the module.
+  * \param[in] base Python name of base class (from pex exceptions).
+  */
+template <typename T>
+void declare_exception(pybind11::module &mod, const std::string & name, const std::string & base) {
+    using lsst::pex::exceptions::Exception;
+
+    // Wrap T as a new Python exception type with *name* and add it to the module
+    //
+    // Note that all created C++ wrapped type derive from Exception here.
+    // It is only in the pure Python wrapper layer that they get embedded in
+    // a subclass of the requested base.
+    pybind11::class_<T> cls(mod, name.c_str(), pybind11::base<Exception>());
```

Comment:
Is there any real difficulty with providing the direct base class here, if we assume it's single inheritance on the C++ side?  Providing another template argument (which could default to `Exception`) seems like it wold make that easy.

I can't think of anything that this simplification would break at the moment, given that the pure-Python wrappers have the right inheritance tree, but I think this would be considered very surprising behavior if it did break something.